src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -163,12 +163,12 @@ private void CheckSourceLocationSeen(AnonymousTypePublicSymbol anonymous)
             }
         }
 
-        internal SynthesizedDelegateSymbol SynthesizeDelegate(int parameterCount, BitVector byRefParameters, bool returnsVoid, int generation)
+        internal SynthesizedDelegateSymbol SynthesizeDelegate(int parameterCount, RefKindVector refKinds, bool returnsVoid, int generation)
         {
             // parameterCount doesn't include return type
-            Debug.Assert(byRefParameters.IsNull || parameterCount == byRefParameters.Capacity);
+            Debug.Assert(refKinds.IsNull || parameterCount == refKinds.Capacity - (returnsVoid ? 0 : 1));
```

Comment:
Did we not excercise `returnsVoid` before? How did we not hit the old assertion?

Reply:
Previously, the return value was assumed to be by value so the `BitVector` represented the parameters only.